### PR TITLE
GRW-1470 / Feature / Unify locale format to xx-XX

### PR DESCRIPTION
**Pull request by alebedev**
## GRW-1470 / Feature / Unify locale format to xx-XX

Remove non-standard locale variants

## Justify why they are needed
Less fields, less normalization, easier to use external APIs

## Checklist before requesting a review

- [x] I have performed a self-review of my code

[GRW-1470]: https://hedvig.atlassian.net/browse/GRW-1470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/store/src/lib/l10n/locales.ts [hunk 1]
```diff
@@ -51,6 +51,14 @@ export enum LocaleField {
   Language = 'language',
 }
 
+export const normalizeLocale = (locale: string | undefined): string | undefined => {
```
Comment (guilhermespopolin) [inline apps/store/src/lib/l10n/locales.ts:54]: Initially I thought about changing `normalizeLocale` so it returns `Locale`. With that we could replace `locales[normalizeLocale(locale) as Locale]` to `locales[normalizeLocale(locale)]`. But that change would trigger so many others I guess it's not worth it 😅

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/lib/l10n/locales.ts [hunk 2]
```diff
@@ -65,6 +51,14 @@ export enum LocaleField {
...
+  if (!locale?.includes('-')) {
+    return locale
```
Comment (gustaveen) [inline apps/store/src/lib/l10n/locales.ts:56]: When would this case be true? 🤔

Comment (alebedev) [reply apps/store/src/lib/l10n/locales.ts:56]: 'default'

File: apps/store/src/lib/l10n/countries.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ const localeCountries = Object.fromEntries(
 ) as Record<Locale, CountryLabel>
 
 export const getCountryByLocale = (locale: string): CountryData => {
-  const countryData = countries[localeCountries[locale as Locale]]
+  const countryData = countries[localeCountries[normalizeLocale(locale) as Locale]]
```
Comment (simonauner) [inline apps/store/src/lib/l10n/countries.ts:43]: Should this be 
```suggestion
  const countryData = countries[localeCountries[getLocale(locale)]]
```
then we don't need to cast it here?

File: apps/store/src/lib/l10n/types.ts [hunk 4]
```diff
@@ -2,12 +2,12 @@ export { CountryCode } from '@/services/apollo/generated'
 
 /** An enum representing explicitly endorsed Locales supported by our system. */
 export enum Locale {
-  SvSe = 'sv-se',
-  EnSe = 'en-se',
-  NbNo = 'nb-no',
-  EnNo = 'en-no',
-  DaDk = 'da-dk',
-  EnDk = 'en-dk',
+  SvSe = 'sv-SE',
```
Comment (simonauner) [inline apps/store/src/lib/l10n/types.ts:5]: gif

File: apps/store/src/lib/l10n/useCurrentLocale.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
+import { FALLBACK_LOCALE, locales, normalizeLocale } from './locales'
 import { Locale } from './types'
 
 export const useCurrentLocale = () => {
   const router = useRouter()
 
   const locale = router.locale === 'default' ? FALLBACK_LOCALE : router.locale || FALLBACK_LOCALE
 
-  return locales[locale as Locale]
+  return locales[normalizeLocale(locale) as Locale]
```
Comment (simonauner) [inline apps/store/src/lib/l10n/useCurrentLocale.ts:10]: `getLocale(locale)` with the same motivation as above?

File: apps/store/src/lib/l10n/countries.ts (hunk 3 above)
Comment (alebedev) [reply apps/store/src/lib/l10n/countries.ts:43]: This may fall back to default locale which is something I'd prefer to avoid here

Comment (simonauner) [reply apps/store/src/lib/l10n/countries.ts:43]: Then I would prefer that we have `getLocaleWithFallback`/`getLocaleOrFallback` and `getLocale` 🤔

File: apps/store/src/lib/l10n/locales.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/lib/l10n/locales.ts:54]: In regular code you just call `getLocale(locale)`, direct locales lookup is only needed if you want special logic when you miss which is few places

**Pull request merged**